core/core.go

```diff
@@ -600,7 +600,7 @@ func (c *Core) WriteBlock(block *types.WorkObject) {
 func (c *Core) Append(header *types.WorkObject, manifest types.BlockManifest, domPendingHeader *types.WorkObject, domTerminus common.Hash, domOrigin bool, newInboundEtxs types.Transactions) (types.Transactions, bool, bool, error) {
 	nodeCtx := c.NodeCtx()
 	// Set the coinbase into the right interface before calling append in the sub
-	header.Header().SetCoinbase(common.BytesToAddress(header.Coinbase().Bytes(), c.NodeLocation()))
+	header.WorkObjectHeader().SetCoinbase(common.BytesToAddress(header.Coinbase().Bytes(), c.NodeLocation()))
```

Comment:
Are we still worried about the coinbase being in the wrong scope?

Reply:
Think so, its still changing the context so, we probably need to make the coinbase internal before sending it tro the sub